src/Raven.Server/Documents/Sharding/Subscriptions/ShardedSubscriptionConnection.cs

```diff
@@ -102,7 +102,7 @@ public override async Task ParseSubscriptionOptionsAsync()
             await base.ParseSubscriptionOptionsAsync();
 
             // we want to limit the batch of each shard, to not hold too much memory if there are other batches while batch is proceed
-            _options.MaxDocsPerBatch = Math.Min(_options.MaxDocsPerBatch / TcpConnection.DatabaseContext.ShardCount, _options.MaxDocsPerBatch);
+            _options.MaxDocsPerBatch = Math.Min((int)Math.Ceiling((double)_options.MaxDocsPerBatch / TcpConnection.DatabaseContext.ShardCount), _options.MaxDocsPerBatch);
```

Comment:
better to wrap all of this in `Math.Max(sth, 1);`

Reply:
done